packages/snippetz/src/core/types-proposed.ts

```diff
@@ -0,0 +1,29 @@
+type NodeClient = 'undici' | 'fetch' | 'ofetch'
+type JSClient = 'fetch' | 'ofetch'
+type ShellClient = 'curl'
+
+type TargetId = 'node' | 'js' | 'shell'
+
+const getTarget = <
+  TTarget extends TargetId,
+  TClient extends TTarget extends 'node'
+    ? NodeClient
+    : TTarget extends 'js'
+      ? JSClient
+      : TTarget extends 'shell'
+        ? ShellClient
+        : never,
+>(
+  target: TTarget,
+  client: TClient,
+): { snippet: string } => {
+  // do snippet things
+  return {
+    snippet: 'my-snippet',
+  }
+}
```

Comment:
Here is an alternative idea for getting type inference in the snippetz. Maybe there are some drawbacks to this approach but I'm preferring this over global. Let's have more discussion!